geomstats/geometry/group_action.py

```diff
@@ -69,6 +70,23 @@ def __call__(self, group_elem, point):
         return self._group.compose(point, group_elt)
 
 
+class LieAlgebraBasedSpecialOrthogonalAction(LieAlgebraBasedGroupAction):
```

Comment:
I don't think the `point_type` (or `coords_type`? 🤔 ) of the group's elements should appear in the name of the class.

Because, in the code base, we usually make the distinction between the math and the way the math is implemented.

Class names <-> Math
The name of the class should refer to the mathematical construct: here, the fundamental action of SOn.

Parameters in class <-> Implementation
We can have a parameter called `point_type`, `coords_type` or else (not sure point_type and coords_type are what we want) that says how the group's points are parameterized.

Similarly to matrix_representation and basis_representation for the points in matrix-spaces.

Reply:
Great point. I'll change accordingly.

Reply:
@ninamiolane, I've added a boolean (`lie_algebra_repr`) to control representation type (if `True`, it uses considers group elements to be represented as vectors of the Lie algebra).

I've also renamed it to `ComposeAction` (probably not a great name). You have a different name to it, right? What is it? Should we rename?